File: app/src/main/java/com/abaffym/moviesapp/ui/detail/MovieDetailViewModel.java
package com.abaffym.moviesapp.ui.detail;

import android.arch.lifecycle.ViewModel;

import com.abaffym.moviesapp.model.Movie;
import com.abaffym.moviesapp.ui.MoviesRepository;

public class MovieDetailViewModel extends ViewModel {

	private MoviesRepository moviesRepository;

	public MovieDetailViewModel() {
		moviesRepository = new MoviesRepository();
	}

	private long movieId = -1;

	public void setMovieId(long movieId) {
		this.movieId = movieId;
	}

	public Movie getMovieDetail() {
		return moviesRepository.getMovieById(movieId);
	}

	public void addToFavorites() {
		moviesRepository.addToFavorites(movieId);
	}

	public void removeFromFavorites() {
		moviesRepository.removeFromFavorites(movieId);
	}

	public boolean isFavorite() {
		return moviesRepository.isFavorite(movieId);
	}
}

File: app/src/main/java/com/abaffym/moviesapp/ui/detail/MovieDetailViewModel.kt
package com.abaffym.moviesapp.ui.detail

import android.arch.lifecycle.ViewModel

import com.abaffym.moviesapp.model.Movie
import com.abaffym.moviesapp.ui.MoviesRepository

class MovieDetailViewModel : ViewModel() {

    private val moviesRepository: MoviesRepository = MoviesRepository()

    private var movieId: Long = -1

    val movieDetail: Movie
        get() = moviesRepository.getMovieById(movieId)

    val isFavorite: Boolean
        get() = moviesRepository.isFavorite(movieId)

    fun setMovieId(movieId: Long) {
        this.movieId = movieId
    }

    fun addToFavorites() = moviesRepository.addToFavorites(movieId)

    fun removeFromFavorites() = moviesRepository.removeFromFavorites(movieId)
}
